### ISSUE TITLE
Can't use old repositories with 2.15.1.windows.2

### ISSUE THREAD

**Issue by sryze** [git-for-windows/git#1430]
When I execute `git diff` or `git rebase` in an existing git repository that worked before I get the following error:

```
fatal: This operation must be run in a work tree
```

This started happening after I reinstalled Windows and consequently reinstalled Git for Windows. I think I had an older version of Git before which worked flawlessly with these repositories. I din't move the repos to another partition or disk, they stayed on the same NTFS partition.

This also happened to me with a newer repo that I cloned after the reinstall. It seems that it somehow got corrupted.

 - [X] I was not able to find an [open](https://github.com/git-for-windows/git/issues?q=is%3Aopen) or [closed](https://github.com/git-for-windows/git/issues?q=is%3Aclosed) issue matching what I'm seeing

### Setup

 - git version 2.15.1.windows.2
 - git version 2.16.0.rc2.windows.1
 - 64-bit

```
$ git --version --build-options

git version 2.15.1.windows.2
built from commit: 5d5baf91824ec7750b103c8b7c4827ffac202feb
sizeof-long: 4
machine: x86_64
```

 - Which version of Windows are you running? Vista, 7, 8, 10? Is it 32-bit or 64-bit?

   Windows 10 Home x64

```
$ cmd.exe /c ver

Microsoft Windows [Version 10.0.16299.192]
```

 - What options did you set as part of the installation? Or did you choose the
   defaults?

```
# One of the following:
> type "C:\Program Files\Git\etc\install-options.txt"
> type "C:\Program Files (x86)\Git\etc\install-options.txt"
> type "%USERPROFILE%\AppData\Local\Programs\Git\etc\install-options.txt"
$ cat /etc/install-options.txt

Editor Option: VIM
Path Option: Cmd
SSH Option: OpenSSH
CURL Option: OpenSSL
CRLF Option: LFOnly
Bash Terminal Option: ConHost
Performance Tweaks FSCache: Enabled
Use Credential Manager: Enabled
Enable Symlinks: Enabled
```

 - Any other interesting things about your environment that might be related
   to the issue you're seeing?

No.

### Details

 - Which terminal/shell are you running Git from? e.g Bash/CMD/PowerShell/other

Git Bash, cmd.exe - same result in both, it doesn't matter

 - What commands did you run to trigger this issue? If you can provide a
   [Minimal, Complete, and Verifiable example](http://stackoverflow.com/help/mcve)
   this will help us understand the issue.

```
git diff
```
 - What did you expect to occur after running these commands?

Show the diff or tell that there is no changes.

 - What actually happened instead?

```
$ git diff
fatal: This operation must be run in a work tree
```

 - If the problem was occurring with a specific repository, can you provide the
   URL to that repository to help us with testing?

No. I can't help with testing though.

**Comment by sryze** [comment]
After some debugging I found that the [`GetFinalPathNameByHandleW()` call in `mingw_chdir`](https://github.com/git-for-windows/git/blob/d0192a7c4e03bb57edcd04ca783ecf349102f549/compat/mingw.c#L696) (which is called by `setup_work_tree`) returns 0 and the error is set to `ERROR_PATH_NOT_FOUND`. Documentation on this function says:

> Can be returned if you are searching for a drive letter and one does not exist. For example, the handle was opened on a drive that is not currently mounted, or if you create a volume and do not assign it a drive letter. If a volume has no drive letter, you can use the volume GUID path to identify it.
>
> This return value can also be returned if you are searching for a volume GUID path on a network share. Volume GUID paths are not created for network shares.

https://msdn.microsoft.com/en-us/library/windows/desktop/aa364962(v=vs.85).aspx

But the drive letter is correct, I can open the same path in Explorer with no problem.

After I removed the drive letter and assigned it again via the Disk Management tool the error is gone. I think this has something to do with the way the drive letter was assigned initially - I did it though a program called Ext2fsd by selecting Change drive letters -> Add -> Create a permanent MountPoint via Session Manager. Apparently this doesn't work well with that Windows funciton
